### PR TITLE
Fix bug that BE crashes when inserting null value to non-nullable columns

### PR DESCRIPTION
**Pull request by chenhao7253886**

### CODE REVIEW

File: be/src/exec/olap_table_sink.cpp [hunk 1]
```diff
@@ -640,14 +640,25 @@ void OlapTableSink::_convert_batch(RuntimeState* state, RowBatch* input_batch, R
...
             RawValue::write(src_val, slot, slot_desc->type(), _output_batch->tuple_data_pool());
         }
+
+        if (!exist_null_value_for_not_null_col) {
+            if (output_batch->add_row() != RowBatch::INVALID_ROW_INDEX) {
+                commit_rows++;
+                output_batch->get_row(i)->set_tuple(0, dst_tuple);
+            } else {
+                // input_batch and output_batch's capacity must be same.
+                DCHECK(false);
```
Comment (morningman) [inline be/src/exec/olap_table_sink.cpp]: You should add ERROR info for this DCHECK.
Like: DCHECK(false) << "input_batch and output_batch's capacity must be same. " << input size << '', " << output size;

File: be/src/exec/olap_table_sink.cpp [hunk 2]
```diff
@@ -640,14 +640,25 @@ void OlapTableSink::_convert_batch(RuntimeState* state, RowBatch* input_batch, R
...
                 }
             }
             void* slot = dst_tuple->get_slot(slot_desc->tuple_offset());
             RawValue::write(src_val, slot, slot_desc->type(), _output_batch->tuple_data_pool());
         }
+
+        if (!exist_null_value_for_not_null_col) {
+            if (output_batch->add_row() != RowBatch::INVALID_ROW_INDEX) {
+                commit_rows++;
+                output_batch->get_row(i)->set_tuple(0, dst_tuple);
```
Comment (morningman) [inline be/src/exec/olap_table_sink.cpp]: it is not right to use `i` as index

**Review by morningman — CHANGES_REQUESTED** [review]

File: be/src/exec/olap_table_sink.cpp (hunk 2 above)
Comment (imay) [inline be/src/exec/olap_table_sink.cpp]: If one row is skipped, `get_row(i)` will get a wrong result. I think you should what you get is add_row()'s return value

**Review by imay — CHANGES_REQUESTED** [review]

File: be/src/exec/olap_table_sink.cpp [hunk 3]
```diff
@@ -640,14 +640,25 @@ void OlapTableSink::_convert_batch(RuntimeState* state, RowBatch* input_batch, R
 #else
                     state->append_error_msg_to_file("", ss.str());
```
Comment (EmmyMiao87) [inline be/src/exec/olap_table_sink.cpp]: please add filtered_rows++;

Comment (chenhao7253886) [reply be/src/exec/olap_table_sink.cpp]: Ok , i will add it.

**Review by imay — APPROVED** [review]
LGTM

**Review by morningman — APPROVED** [review]

**Pull request merged**